api/src/modules/scenarios-planning-unit/entities/scenarios-planning-unit.geo.entity.ts

```diff
@@ -0,0 +1,72 @@
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+import { LockStatus } from '../lock-status.enum';
+
+export const scenariosPuDataEntityName = 'scenarios_pu_data';
+
+const toLockEnum: Record<1 | 2, LockStatus> = Object.freeze({
+  1: LockStatus.LockedIn,
+  2: LockStatus.LockedOut,
+});
+
+const fromLockEnum: Record<LockStatus, 1 | 2 | null> = Object.freeze({
+  [LockStatus.Unknown]: null,
```

Comment:
I think we should also explicitly handle the value `0` in the _from_ transform as this is something we'll normally want to set a PU `lockin_status` to for any PUs which are not explicitly included or excluded (and we then use this value of `0` verbatim when generating Marxan .dat files - ofc we could as well just store `1 | 2 | null` in the column and only output the `0`s when generating .dat files, but I'd keep the initial design on this).

Status `0` thus becomes something that may be labelled "not locked" or "unstated" (or something similar - I think NotLocked is good enough).

Reply:
@hotzevzl sure, I just assumed we don't use `0`
![image](https://user-images.githubusercontent.com/3466388/117027534-15f4cc00-acfd-11eb-8f9e-46cb1febd7ae.png)
(will fix!)

Reply:
@hotzevzl I would keep using `unknown` we already have there (their meaning is the same, isn't it?)

Reply:
on the other hand, if we want to keep `null` instead of `0` (`null` is possible to provide) this would require a bit of mess (i.e. `lockStatus!: LockStatus | null;`)

Reply:
I think we should either allow null and restrict the non-null values to 1, 2 or 3 (we don't use 3 in the app, but it's a legitimate value for Marxan data, and we will work on importing legacy Marxan projects later on, which could use this value, in which case we'd map 0 in the input to null in the db), *or* make the column non-nullable with default 0 and possible values restricted to 0, 1, 2 and 3.
I'd defer this decision to Alicia though, as she will have other implications of any such choices in her plans.

The name is really just minor semantics, and since we don't surface this to the users it's largely immaterial. In practice, 0 in Marxan means that the PU is neither locked in nor locked out, that means it's available for a decision on its final status (included or excluded from the proposed planning solution) by the Marxan CLI's algorithm, so strictly speaking is "not locked" (whereas a PU which is locked in is always guaranteed to be tagged as included in the planning solution).

Reply:
updated in https://github.com/Vizzuality/marxan-cloud/pull/148/commits/6396c14d671e06fbc11259fc539499d536454a85

Decided to keep "current" values (null,0,1) until we do a proper migration according to Alicia's decision.

I also put jsdoc for few mentioned values for future reference.

Reply:
First of all, let me put here the values that the lock-in column should be:

```
{
                       0: 'The PU is not guaranteed to be in the initial reserve',
                       1: 'The PU will be included in the initial reserve',
                       2: 'The PU is fixed in the reserve system (“locked-in”). starts in the initial reserve system and cannot be removed.',
                       3: 'The PU is fixed outside the reserve system (“locked-out”). It is not included in the initial reserve system and cannot be added.'
                      }
```

So my recommendation: In this column, we should make the column non-nullable with default 0 and possible values restricted to 0, 1, 2, and 3 as Andrea suggested. Making them nullable will require us to make a transformation at `.dat ` creation or project import that I think we can avoid. Most of the projects will go or 0 or 2 or 3 in very rare cases.

Reply:
@aagm the issue is that it is currently allowed to put `null`s. I am fine to adjust them but this may require adjustments on GeoService side. How about making it ina separate PR and touch all places where it is needed?

Reply:
sounds good to me, lets migrate this on a separate pr

Reply:
https://vizzuality.atlassian.net/browse/MARXAN-398